src/master-playlist-controller.js

```diff
@@ -429,8 +402,17 @@ export class MasterPlaylistController extends videojs.EventTarget {
       let updatedPlaylist = this.masterPlaylistLoader_.media();
 
       if (!updatedPlaylist) {
-        // select the initial variant
-        this.initialMedia_ = this.selectPlaylist();
+        let selectedMedia;
```

Comment:
That loop seems like a code smell to me, though I can't think of a better way to deal with it in the current context.

I believe I handled the scenario in your comment below with the latest commit.